locale/en/docs/guides/dont-block-the-event-loop.md

```diff
@@ -0,0 +1,457 @@
+
+However, if your server relies heavily on complex calculations, you should think about whether Node is really a good fit. Node excels for I/O-bound work, but for expensive computation it might not be the best option.
+
+If you take the offloading approach, see the section on not blocking the Worker Pool.
+
+## Don't block the Worker Pool
+Node has a Worker Pool composed of `k` Workers.
+If you are using the Offloading paradigm discussed above, you might have a separate Computational Worker Pool, to which the same principles apply.
+In either case, let us assume that `k` is much smaller than the number of clients you might be handling concurrently.
+This is in keeping with Node's "one thread for many clients" philosophy, the secret to its scalability.
+
+As discussed above, each Worker completes its current Task before proceeding to the next one on the Worker Pool queue.
+
+Now, there will be variation in the cost of the Tasks required to handle your clients' requests.
+Some Tasks can be completed quickly (e.g. reading short or cached files, or producing a small number of random bytes), and others will take longer (e.g reading larger or uncached files, or generating more random bytes).
+Your goal should be to *minimize the variation in Task times*, and you should use *Task partitioning* to accomplish this.
+
+### Minimizing the variation in Task times
+If a Worker's current Task is much more expensive than other Tasks, then it will be unavailable to work on other pending Tasks.
+In other words, *each relatively long Task effectively decreases the size of the Worker Pool by one until it is completed*.
+This is undesirable because, up to a point, the more Workers in the Worker Pool, the greater the Worker Pool throughput (tasks/second) and thus the greater the server throughput (client requests/second).
+One client with a relatively expensive Task will decrease the throughput of the Worker Pool, in turn decreasing the throughput of the server.
+
+To avoid this, you should try to minimize variation in the length of Tasks you submit to the Worker Pool.
+While it is appropriate to treat the external systems accessed by your I/O requests (DB, FS, etc.) as black boxes, you should be aware of the relative cost of these I/O requests, and should avoid submitting requests you can expect to be particularly long.
+
+Two examples should illustrate the possible variation in task times.
+
+#### Variation example: Long-running file system reads
+Suppose your server must read files in order to handle some client requests.
+After consulting Node's [File system](https://nodejs.org/api/fs.html) APIs, you opted to use `fs.readFile()` for simplicity.
+However, `fs.readFile()` is not partitioned: it submits a single `fs.read()` Task spanning the entire file.
```

Comment:
This is actually being fixed if I understand correctly: https://github.com/nodejs/node/pull/17054

Reply:
That's the idea, anyway. But this guide is describing things as they are, not as I hope they will be.

Reply:
See also nodejs/node#17250.

Reply:
Might be worth adding a link to nodejs/node#17054 in here to note the proposed change then.

Reply:
Added a link in 0a42816, thank you.